app/views/layouts/application.html.erb

```diff
@@ -21,18 +21,17 @@
 <% if user_signed_in? %>
   <%= content_for :navbar_right do %>
     <div class='session-details-panel'>
-      <span class='session-details-panel__username'>Logged in as: <%= current_user.name %></span>
-      <%= link_to 'Logout', destroy_user_session_path %>
-      <%= link_to 'Change password', edit_user_registration_path %>
+      <%= link_to current_user.name, Plek.current.find('signon') %>
+      &bull; <%= link_to 'Sign out', gds_sign_out_path %>
     </div>
   <% end %>
 
-  <% if current_user.team_leader? %>
+  <% if current_user.has_permission?('team_leader') %>
```

Comment:
These might as well be the same predicates as before. Seems a little cleaner to keep the role name hidden away as knowledge of the user.